flow/connectors/clickhouse/clickhouse.go

```diff
@@ -542,3 +543,63 @@ func (c *ClickHouseConnector) GetVersion(ctx context.Context) (string, error) {
 	c.logger.Info("[clickhouse] version", slog.Any("version", clickhouseVersion.DisplayName))
 	return clickhouseVersion.Version.String(), nil
 }
+
+func (c *ClickHouseConnector) getTableSchemaForTable(ctx context.Context, tableName string) (*protos.TableSchema, error) {
+	// TODO sanitize
+	q, err := c.database.Query(ctx, fmt.Sprintf("select * from %s limit 0", tableName))
+	if err != nil {
+		return nil, err
+	}
+
+	columns := q.ColumnTypes()
+	colFields := make([]*protos.FieldDescription, 0, len(columns))
+	for _, column := range columns {
+		var qkind qvalue.QValueKind
+		switch column.DatabaseTypeName() {
```

Comment:
would prefer to lift this into a function so it can be reused here and in `.GetRows`